server/models/Share.js

```diff
@@ -44,6 +44,27 @@ Share.associate = (models) => {
       { association: "team" },
     ],
   });
+  Share.addScope("withCollection", (userId) => {
+    return {
+      include: [
+        {
+          model: models.Document,
+          paranoid: true,
+          as: "document",
+          include: [
+            {
+              model: models.Collection.scope({
+                method: ["withMembership", userId],
+              }),
+              as: "collection",
+            },
+          ],
+        },
+        { association: "user", paranoid: false },
+        { association: "team" },
```

Comment:
I was confused about whether to include these associations or not. In the end, took the decision to include otherwise we will have to scope `defaultScope` and `withCollection` scope in each query.

There is another possible solution to include collection memberships in the default scope but the fetched memberships will be left unused.